src/pages/NewChairConsole.js

```diff
@@ -151,6 +190,30 @@ const NewChairConsole = () => {
     }
   };
 
+  const createElection = (darId, index) => {
+    if (darId !== null) {
+      let copy;
+      Election.createDARElection(darId)
+        .then(
+          Notifications.showSuccess({text: "Election successfully opened"}),
+          copy = cloneDeep(filteredList),
+          copy[index].election = { status: "Open", finalAccessVote: false },
```

Comment:
Two key notes here, bit long but it's important to read through it:

1. We don't want to overwrite keys on objects directly unless we know we're not removing key data. If copy[index].election was something simple like a string or an int then we can overwrite it with no hesitation. But if it's something like an object we need to think about the other keys that can exist within it. So if the election looked like this...
```
{
  name: 'Election Name',
  id:273,
  status: 'Closed',
  finalAccessVote: false
}
```
... then we're going to lose the name and the id on the election, since we are now setting election to only ```{status: "Open", finalAccessVote: false}```. So instead of overwriting the election completely, you can do something like...
```
  copy[index].election.status = "Open";
  copy[index].finalAccessVote = false;
```
...to update the values we want.


2. Even if we do the above, nothing will update on the screen!

![Kapture 2021-02-11 at 09 49 50](https://user-images.githubusercontent.com/6664778/107654114-55f16900-6c50-11eb-96c5-9a0b05df17ac.gif)

That's because React updates the screen by comparing the references of variables! It does this to optimize re-renders since it's faster to compare references than it is to compare every single key on an object (which can be very broad or, if it has a bunch of nested keys, very deep). With the above method, even though we've changed ```election.status```, election itself is still the same. To make a copy and update the attributes, we can use Object.assign(). Here's an example...
```
copy[index].election = Object.assign({}, copy[index].election, {status: "Open", finalAccessVote: false}
```

The way it works is that, iterating from right to left, apply the keys of the current object to the object on the left. If both objects have the same keys, then the value of the current object will be used to "update" the value of the object to the left. (I say "update" because the only object that has its value changed is the left most object). We use the empty object on the left because it's a fresh empty object, which means that it'll have a new reference (the second condition we're aiming for). Once we do the above, the ```setFilteredList(copy)``` should work as expected!

NOTE: lodash has it's own assign function that you can use that has the same result, but performs the operation differently (applies from left to right, always returns a copy so you don't need to add an empty object, etc.) Feel free to use whichever one you want, just note that it's important to be aware of implementation differences when using third-party libraries.

Lodash assign docs -> https://lodash.com/docs/4.17.15#assign

Reply:
Okay, thank you for this explanation!!! That makes sense about tracking variable references instead! The reason I was overwriting is because the election may not exist yet in the case that the status is unreviewed, so we can't always assign direct field values, also this is the same overwriting that happens in createDARElection so I was thinking we would want the elections to match in both places. But, looking at the Object.assign() line you have written above, that would work even if copy[index].election is null, right? the new object would just have those two properties, and if it isn't null the new object will have the additional data as well

Reply:
Yep, exactly!